pkg/api/store/storageclass/store_test.go

```diff
@@ -0,0 +1,79 @@
+package storageclass
+
+import (
+	"testing"
+
+	"github.com/rancher/norman/types"
+	"k8s.io/utils/pointer"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type dummyStore struct{}
+
+func createWithFields(store types.Store, prov string, kind string, storageaccounttype *string) (map[string]interface{}, error) {
+	testData := map[string]interface{}{
+		"provisioner": prov,
+		"parameters": map[string]interface{}{
+			"kind": kind,
+		},
+	}
+
+	if storageaccounttype != nil {
+		testData["parameters"].(map[string]interface{})["storageaccounttype"] = *storageaccounttype
+	}
+
+	return store.Create(nil, nil, testData)
+}
+
+func TestCreate(t *testing.T) {
+	assert := assert.New(t)
+	testStore := Wrap(dummyStore{})
+
+	// create with a storageaccounttype of nil should give the storage class a default value for the field
+	data, err := createWithFields(testStore, AzureDisk, "shared", nil)
+	assert.Nil(err)
+	assert.Equal("Standard_LRS", data["parameters"].(map[string]interface{})["storageaccounttype"])
+
+	// create with a kind that is not shared should not make a change to the storageaccounttype
+	data, _ = createWithFields(testStore, AzureDisk, "dedicated", nil)
+	assert.Nil(err)
```

Comment:
The err is dropped but you are still asserting it, should just not drop the error.